src/plot_api/plot_api.js

```diff
@@ -1686,6 +1686,74 @@ function _restyle(gd, aobj, traces) {
     };
 }
 
+/**
+ * Converts deprecated layout attributes to the current API
+ * to ensure backwards compatibility.
+ *
+ * @param layoutObj
+ */
+function cleanDeprecatedLayoutAttributes(layoutObj) {
+
+    // Support old-style update of the title's font
+    var isTitlefontSet = layoutObj.titlefont;
+    var isFontInTitleSet = layoutObj.title && layoutObj.title.font;
+    var isFontInTitleNotSet = !isFontInTitleSet;
+    if(isTitlefontSet && isFontInTitleNotSet) {
+
+        // Use string attribute because initiating a new title object
+        // to be able to specify a font property on it would require to
+        // know the potentially existing `title.text` property.
+        layoutObj['title.font'] = layoutObj.titlefont;
+        delete layoutObj.titlefont;
+    }
+
+    // Note, that updating x-axis and y-axis title fonts
+    // was never supported unless (i) using string
+    // attributes or (ii) passing `xaxis.title` / `yaxis.title`
+    // again. And these cases are covered by
+    // helpers.cleanLayout anyways.
+}
+
+/**
+ * Converts deprecated layout "string attributes" to
+ * "string attributes" of the current API to ensure backwards compatibility.
+ *
+ * E.g. Maps {'xaxis.title': 'A chart'} to {'xaxis.title.text': 'A chart'}
+ *
+ * @param layoutObj
+ */
+function cleanDeprecatedLayoutAttributeStrings(layoutObj) {
+    var oldAxisTitleRegExp = /axis\d{0,2}.title$/;
+    var keys, i, key, value;
+
+    if(typeof layoutObj.title === 'string') {
+        layoutObj.title = {text: layoutObj.title};
+    }
+
+    // Note: Only "nested" (dot notation) attributes
+    // need to be converted. For example 'layout.titlefont'
+    // was a top-level attribute and thus is covered by
+    // the general compatibility layer.
+    keys = Object.keys(layoutObj);
+    for(i = 0; i < keys.length; i++) {
+        key = keys[i];
+        value = layoutObj[key];
+
+        if(key.indexOf('titlefont') > -1) {
+            replace(key, key.replace('titlefont', 'title.font'));
+        }
+
+        if(typeof value === 'string' && oldAxisTitleRegExp.test(key)) {
```

Comment:
Here too, we have to accept more than just `'string'` types.

Reply:
Yep!